File: Demo/OC/AppFrameworks/Foundation/YJFoundation/YJFoundationTests/URLSessionTest.m
//
//  URLSessionTest.m
//  YJFoundationTests
//
//  Created by 阳君 on 2019/6/21.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

#import <XCTest/XCTest.h>
#import <YJCocoa/YJCocoa.h>
#import <YJCocoa/YJCocoa-Swift.h>
#import "YJFoundationTests-Swift.h"

@interface URLSessionTest : XCTestCase
@end
@implementation URLSessionTest

- (void)testURLSession {
    YJTestURLRequestModel *requestModel = [[YJTestURLRequestModel alloc] init];
    requestModel.name = @"阳君";
    requestModel.qq = @"557445088";
    YJURLRequest *request = [[YJURLRequest alloc] initWithSource:nil url:@"https://github.com/937447974/YJCocoa" method:YJURLRequestMethodPost reqModel:requestModel respModel:YJTestURLResponseModel.new];
   [[[YJTestURLSessionTask taskWith:request] completionHandlerWithSuccess:^(YJTestURLResponseModel *respModel) {
         NSLog(@"获取服务器数据:%@", respModel.desc);
    } failure:^(NSError * error) {
        NSLog(@"%@", error);
    }] resume];
}

@end

File: Demo/Swift/AppFrameworks/Foundation/YJFoundation/YJFoundationTests/URLSessionTest.swift
//
//  URLSessionTest.swift
//  YJFoundationTests
//
//  Created by 阳君 on 2019/6/17.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import XCTest
import YJCocoa

class URLSessionTest: XCTestCase {
    
    struct YJTestURLRequestModel {
        var name: String
        var qq: String
    }
    
    struct YJTestURLResponseModel {
        var desc: String?
    }
    
    class YJTestURLSessionTask: YJURLSessionTask {
        override func resume() {
            guard self.state != .running else {
                return
            }
            super.resume()
            dispatch_after_default(delayInSeconds: 1) {
                let json = "{\"desc\": \"服务器回调\"}"
                self.success(json)
            }
        }
    }
    
    func testURLSession() {
        let reqModel = YJTestURLRequestModel(name: "阳君", qq: "557445088")
        let reqDict = YJJSONModel.transformToDict(reqModel)
        let request = YJURLRequest(source: nil, url: "https://github.com/937447974/YJCocoa", method: .post, reqModel: reqDict, respModel: YJTestURLResponseModel())
        let task = YJTestURLSessionTask.task(with: request)
        task.completionHandler(success: { (respModel) in
            print(respModel)
        }, failure: { (error) in
            print("错误")
        }).resume()
    }
    
    func testQueue() {
        for _ in 0..<30 { //合并网络请求
            self.testURLSession()
        }
        sleep(2)
    }
    
}
